File: src/0806.number-of-lines-to-write-string.824/solution.js
// 水题
/**
 * @param {number[]} widths
 * @param {string} S
 * @return {number[]}
 */
var numberOfLines = function (widths, S) {
    let lineCount = 1;
    let curLine = 0;
    for (let i = 0; i < S.length; i++) {
        const width = widths[S.charCodeAt(i) - 97];
        if (curLine + width < 101) {
            // 还能放得下，更新占据的空间
            curLine += width;
        } else {
            // 放不下了，换行
            lineCount++;
            curLine = width;
        }
    }
    return [
        lineCount, curLine, ];
};

File: src/0806.number-of-lines-to-write-string.824/solution.java
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int row = 0;
        int col = 0;
        for(char c : s.toCharArray()){
            int w = widths[c-'a'];
            if(col+w>100){
                row++;
                col = w;
            }else{
                col += w;
            }
        }
        return new int[]{row+1,col};
    }
}